libclingo/clingo.h

```diff
@@ -2057,10 +2079,21 @@ CLINGO_VISIBILITY_DEFAULT bool clingo_statistics_map_subkey_name(clingo_statisti
 //! @param[in] name the name to lookup the subkey
 //! @param[out] subkey the resulting subkey
 //! @return whether the call was successful
-CLINGO_VISIBILITY_DEFAULT bool clingo_statistics_map_at(clingo_statistics_t *statistics, uint64_t key, char const *name, uint64_t *subkey);
+CLINGO_VISIBILITY_DEFAULT bool clingo_statistics_map_at(const clingo_statistics_t *statistics, uint64_t key, char const *name, uint64_t *subkey);
+//! Create a subkey under the given name.
+//!
+//! @pre The @link clingo_statistics_type() type@endlink of the entry must be @ref ::clingo_statistics_type_map.
+//! @param[in] statistics the target statistics
+//! @param[in] key the key
+//! @param[in] name the name of the new subkey
+//! @param[in] type the type of the new subkey
+//! @param[out] subkey the resulting subkey
+//! @return whether the call was successful
+CLINGO_VISIBILITY_DEFAULT bool clingo_statistics_map_create(clingo_statistics_t *statistics, uint64_t key, char const *name, clingo_statistics_type_t type, uint64_t *subkey);
```

Comment:
I do not like the name `map_create`. I think `map_add_subkey` would better fit with the existing interface.